### ISSUE TITLE
User should enable/disable multiple addons at once

### ISSUE THREAD

**Issue by budhrg** [minishift/minishift#665]
This is again a better user experience IMO for addon where user should enable/disable multiple addons through single commands.
ATM, it is just one addon per command.

```
$ minishift addons list                          
- anyuid         : disabled   P(0)
- cluster-admin  : disabled   P(0)
- registry       : disabled   P(0)

$ minishift addons enable anyuid cluster-admin registry

$ minishift addons list                                
- anyuid         : enabled    P(0)
- cluster-admin  : disabled   P(0)
- registry       : disabled   P(0)
```

We can also implement same for `install` subcommand though we are also looking to install multiple addons based on folder, see here https://github.com/minishift/minishift/issues/635#issuecomment-291547294.

**Comment by praveenkumar** [comment]
@budhrg I agree that it's a better user experience but I don't think it should have priority as `major` because we are not breaking any usecase here.

**Comment by budhrg** [comment]
Updated @praveenkumar 👍

**Comment by hferentschik** [comment]
And how do you handle priorities?

**Comment by hferentschik** [comment]
BTW, where is this issue coming from and why is it added to rc.2?

**Comment by LalatenduMohanty** [comment]
The tricky part if how we handle the priorities when we enable multiple add-ons at one go. I think we have to assume that when we enable multiple add-ons , the add-ons have same priority. WDYT?

**Comment by hferentschik** [comment]
>  I think we have to assume that when user enable multiple add-ons , the add-ons have same priority.

I guess.

**Comment by gbraad** [comment]
for the moment yes... priorities can also be solved by dependency resolving (but this involves pre-processing)

**Comment by praveenkumar** [comment]
I think we should drop this feature right now or if we implement then we need each field mandatory (addon name, priority etc.) Problem is if we enable this features then user might use it for different priority addons and then again hit the issue.

**Comment by budhrg** [comment]
> The tricky part if how we handle the priorities when we enable multiple add-ons at one go. I think we have to assume that when user enable multiple add-ons , the add-ons have same priority. WDYT?

Yes. Agree. Also, depend on how much far we go with this feature too. As @gbraad  said, dependency resolving is key here.

**Comment by hferentschik** [comment]
> I think we should drop this feature right now

Works for me

> or if we implement then we need each field mandatory (addon name, priority etc.) 

not sure what you mean, but I guess you are referring to the fact on how to enable multiple add-ons with different priorities. I don't see a problem here. If you want to add multiple add-ons with different priority you need to make multiple calls (one for each priority). I don't see an issue here really. 

> Problem is if we enable this features then user might use it for different priority addons and then again hit the issue.

I am not following 

> As @gbraad said, dependency resolving is key here.

I don't see dependency resolving as a necessity. There is a feature request for explicitly named priorities, but I don't think this is required and makes things just more complicated.

**Comment by praveenkumar** [comment]
> not sure what you mean, but I guess you are referring to the fact on how to enable multiple add-ons with different priorities. I don't see a problem here. If you want to add multiple add-ons with different priority you need to make multiple calls (one for each priority). I don't see an issue here really.

@hferentschik right, I meant if we want to implement this feature which should also handle priority as part of enable multiple addons as single go then user have to specify name and priority of addon.
Something like `minishift addons enable A <priority_A> B <priority_B> ...etc.`

**Comment by praveenkumar** [comment]
> I think we have to assume that when user enable multiple add-ons , the add-ons have same priority. 

If this is what we want to implement then I am not sure how much value we will add for users.  what if a user have lot of addons with a specific priority then does he able to use this way? If we want to have this only for same priority then there should be a option to mention priority for this addon then default. Something like `minishift addons enable A B C <priority_addon>`

**Comment by jorgemoralespou** [comment]
Why not adding the order you enable the addons as a relative priority? Also, the order in which they appear in the config file, so they can just be moved up/down in the execution chain. 

At the end, addons will not be executed in parallel, but rather sequentially, so why not chaining them, and if I define:

addons apply anyuid cluster-admin cors sample-app 

that will be the order they get executed?

**Comment by LalatenduMohanty** [comment]
> Why not adding the order you enable the addons as a relative priority? Also, the order in which they appear in the config file, so they can just be moved up/down in the execution chain.

@jorgemoralespou That will not work if user already has some add-ons with different priority and has conflicting priorities with the new ones.

**Comment by jorgemoralespou** [comment]
@LalatenduMohanty correct.
For the apply command, what I mention can be doable, as they will be only applied those passed as command line.
For the enable command, you're right, and priority could be needed.

If priority is provided, I would like the addons to be listed in priority or name, so add a flag to the "addons list" that can tell what field will order. By default name is fine.

**Comment by LalatenduMohanty** [comment]
Does not as a priority at this point of view. Keeping it as closed.
